File: queue/Python/Queue.py
class Queue:
	def __init__(self):
		self.element = []

	def enqueue(self, el):
		self.element.insert(len(self.element), el)

	def dequeue(self):
		return self.element.pop(0)

	def empty(self):
		return self.element == []

	def length(self):
		return len(self.element)

	def end(self):
		return self.element[-1] 

	def front(self):
		return self.element[0]

new_queue = Queue()

# Insert 1, 2, 3 to a queue
new_queue.enqueue(1)
new_queue.enqueue(2)
new_queue.enqueue(3)

# Should return True 
print 'Is empty: ', new_queue.empty()

# Displays element in front of the queue
print 'first element: ', new_queue.front()

# Displays element at the end of the queue
print 'last element: ', new_queue.end()

# Remove each element one by one
new_queue.dequeue()
new_queue.dequeue()
new_queue.dequeue()

# Should return False
# since the queue is already empty
print 'Is empty: ', new_queue.empty()

File: queue/java/queue.java
import java.util.NoSuchElementException;
public class queue<Item>{
    private int size; // size of the Stack
    private Node first;
    private Node last; // top of Stack

    private class Node {
        private Item item;
        private Node next;
    }

    public queue() {
        first = null;
        last = null;
        size = 0;
    }

    public boolean isEmpty() {
        return first == null;
    }

  public int getSize(){
    return size;
  }

  public void enqueue(Item item){
    // Add item to the end of the list.
    Node oldlast = last;
    last = new Node();
    last.item = item;
    last.next = null;
      if (isEmpty()) first = last;
      else oldlast.next = last;
    size++;
  }
  public Item dequeue()throws  NoSuchElementException{
    // Remove item from the beginning of the list.
    if(isEmpty())throw new NoSuchElementException("queue underflow")
    Item item = first.item;
    first = first.next;
    if (isEmpty()) last = null;
    size--;
    return item;
  }
  public static void main(String[] args){
     // Create a queue and enqueue/dequeue strings.
     Queue<String> q = new Queue<String>();
    q.enqueue("I am the first one");
    q.enqueue("I am the second one ");
    q.enqueue("I am the third one ");
    System.out.print(q.dequeue());//I am the first one
    System.out.print(q.dequeue());//I am the second one
    System.out.print(q.dequeue());//I am the third one

}
